modules/payment/src/Plugin/Commerce/PaymentGateway/OffsitePaymentGatewayInterface.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Drupal\commerce_payment\Plugin\Commerce\PaymentGateway;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+
+/**
+ * Defines the base interface for off-site payment gateways.
+ */
+interface OffsitePaymentGatewayInterface extends PaymentGatewayInterface {
+
+  /**
+   * Gets the URL for offiste redirect cancel.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   *
+   * @return \Drupal\Core\Url
+   *   The Url object
+   */
+  public function getPaymentRedirectCancelUrl(OrderInterface $order);
+
+  /**
+   * Gets the URL for offiste redirect return.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   *
+   * @return \Drupal\Core\Url
+   *   The Url object
+   */
+  public function getPaymentRedirectReturnUrl(OrderInterface $order);
+
+  /**
+   * Invoked when the off-site payment return.
+   *
+   * @param \Drupal\commerce_order\Entity\OrderInterface $order
+   *   The order.
+   */
+  public function onRedirectReturn(OrderInterface $order);
```

Comment:
I made those changes except Return/Success. Just because the payment gateway returned doesn't mean it was successful, does it? It just means that payment was submitted and returned and the transaction can be processed.

I was wondering same thing about naming in 1.x, but it makes sense that there could be other processing See PayPal code for 1.x

```
  // If we didn't get an approval response code...
  switch ($ipn['payment_status']) {
    case 'Failed':
      $transaction->status = COMMERCE_PAYMENT_STATUS_FAILURE;
      $transaction->message = t("The payment has failed. This happens only if the payment was made from your customer’s bank account.");
      break;

    case 'Voided':
      $transaction->status = COMMERCE_PAYMENT_STATUS_FAILURE;
      $transaction->message = t('The authorization was voided.');
      break;

    case 'Pending':
      $transaction->status = COMMERCE_PAYMENT_STATUS_PENDING;
      $transaction->message = commerce_paypal_ipn_pending_reason($ipn['pending_reason']);
      break;

    case 'Completed':
      $transaction->status = COMMERCE_PAYMENT_STATUS_SUCCESS;
      $transaction->message = t('The payment has completed.');
      break;

    case 'Refunded':
      $transaction->status = COMMERCE_PAYMENT_STATUS_SUCCESS;
      $transaction->message = t('Refund for transaction @txn_id', array('@txn_id' => $ipn['parent_txn_id']));
      break;
  }
```